openPMD: add parameter `particleIOChunkSize` (#4874)
ComputationalRadiationPhysics/picongpu: docs/source/usage/plugins/openPMD.rst

``--checkpoint.openPMD.jsonRestart``  Set backend-specific parameters for openPMD backends in JSON format for restarting from a checkpoint.
``--openPMD.dataPreparationStrategy`` Strategy for preparation of particle data ('doubleBuffer' or 'mappedMemory'). Aliases 'adios' and 'hdf5' may be used respectively.
``--openPMD.toml``                    Alternatively configure the openPMD plugin via a TOML file (see below).
``--openPMD.particleIOChunkSize``     Particle data will be written in chunks of the given size. unit: MiB
                                      The memory footprint on the host side is reduced compared to writing all particles with one write call.

>                                       Currently the memory footprint is only reduced bp5 is used, HDF5 and bp4 does not support partial data dump.

Reviewer: ```suggestion
                                      Currently the memory footprint is only reduced if bp5 is used, HDF5 and bp4 does not support partial data dump.
```

>                                       Currently the memory footprint is only reduced bp5 is used, HDF5 and bp4 does not support partial data dump.

Reviewer 2: This is not a sentence. Is an if missing? 
```suggestion
                                      Currently the memory footprint is only reduced if bp5 is used, HDF5 and bp4 does not support partial data dump.
```

Reviewer 2: Discussed *offline* with @psychocoderHPC: the typo will be fixed in a followup pll request.

During I/O, particle attributes are allocated one after another.
Using ``--openPMD.dataPreparationStrategy doubleBuffer`` (default) will require at least 2x the GPU memory on the host side.
For a smaller host side memory footprint (<< GPU main memory) pick ``--openPMD.dataPreparationStrategy mappedMemory``.

> By using additional to the mapped memory strategy ``--openPMD.particleIOChunkSize N`` you can reduce the memory footprint for particle IO to less than ``2*N`` MiB.

Reviewer: ```suggestion
In addition to using the mapped-memory strategy, you can reduce the memory footprint for particle IO to less than ``2*N`` MiB ``--openPMD.particleIOChunkSize N``.
```
